File: 백준/Silver/1181. 단어 정렬/단어 정렬.py
import sys

n = int(sys.stdin.readline())
arr = [""] * n

for i in range(n):
  arr[i] = sys.stdin.readline()

arr = list(set(arr))
arr.sort()
arr.sort(key = lambda x : len(x))

s = ""
for i in range(len(arr)):
  s += (arr[i])

sys.stdout.write(s)

File: 백준/Silver/1181. 단어 정렬/단어 정렬.cc
#include <iostream>
#include <algorithm>
using namespace std;

int cmp(string a, string b) {
  if (a.length() == b.length())
    return a < b;
  else
    return a.length() < b.length();
}

string arr[20001];

int main() {
  int n;
  cin >> n;

  for (int i = 0; i < n; i++)
    cin >> arr[i];

  sort(arr, arr + n, cmp);

  for (int i = 0; i < n; i++) {
    if (arr[i] == arr[i-1])
      continue;
    
    cout << arr[i] << "\n";
  }
}